Index: linux-7.0/drivers/gpu/drm/drm_edid.c
===================================================================
--- linux-7.0.orig/drivers/gpu/drm/drm_edid.c
+++ linux-7.0/drivers/gpu/drm/drm_edid.c
@@ -2179,7 +2179,10 @@ drm_do_probe_ddc_edid(void *data, u8 *bu
 		 * Avoid sending the segment addr to not upset non-compliant
 		 * DDC monitors.
 		 */
-		ret = i2c_transfer(adapter, &msgs[3 - xfers], xfers);
+		if (!adapter)
+			ret = -ENXIO;
+		else
+			ret = i2c_transfer(adapter, &msgs[3 - xfers], xfers);
 
 		if (ret == -ENXIO) {
 			DRM_DEBUG_KMS("drm: skipping non-existent adapter %s\n",
Index: linux-7.0/drivers/gpu/drm/drm_edid.c
===================================================================
--- linux-7.0.orig/drivers/gpu/drm/drm_edid.c
+++ linux-7.0/drivers/gpu/drm/drm_edid.c
@@ -3525,5 +3525,10 @@ static void drm_mode_do_interlace_quirk(struct drm_display_mode *mode,
 		{ 2880,  480 },
 		{  720,  576 },
 		{ 1440,  576 },
 		{ 2880,  576 },
+		{  800,  600 },
+		{ 1792, 1344 },
+		{ 1600, 1200 },
+		{ 2128, 1596 },
+		{ 1920, 1440 },
 	 };
